crates/bin/prove_block/src/state_utils.rs

```diff
@@ -14,9 +16,21 @@ use starknet_os::starknet::business_logic::fact_state::state::SharedState;
 use starknet_os::starkware_utils::commitment_tree::base_types::TreeIndex;
 // trait for Patricia Tree
 use starknet_os::starkware_utils::commitment_tree::binary_fact_tree::BinaryFactTree;
+use starknet_os::starkware_utils::commitment_tree::patricia_tree::patricia_tree::PatriciaTree;
 use starknet_os::storage::storage::FactFetchingContext;
 
-use crate::rpc_utils::CachedRpcStorage;
+use crate::rpc_utils::{CachedRpcStorage, RpcStorage};
+use crate::utils::get_subcalled_contracts_from_tx_traces;
+
+#[derive(Clone)]
+pub struct ProcessedStateUpdate {
```

Comment:
What is a "processed" state update?

Reply:
Its a "SharedState compatible" set of updates. 
Maybe 'FormattedStateUpdate' ?

Reply:
Yeah, something like that would be clearer.